hendrics/efsearch.py

```diff
@@ -806,11 +807,16 @@ def search_with_qffa_step(
     else:
         phases = _fast_phase(times, mean_f)
 
+    # If weights are not provided, assume uniform weights of 1
+    if weights is None:
```

Comment:
If I'm not mistaken, the algorithm should work even if you keep the `weight` keyword to `None`. If you force it to be `np.ones`, you add an additional multiplication in all steps that most of the time will be unneeded, making the algorithm a lot less efficient in the typical use case.